src/typeDefs/mutations.ts

```diff
@@ -200,6 +200,10 @@ export const mutations = gql`
 
     unregisterForEventByUser(id: ID!): Event! @auth
 
+    updateAdvertisement(
+      input: UpdateAdvertisementInput!
+    ): UpdateAdvertisementPayload! @auth
```

Comment:
return type should be nullable so that error propagation doesn't nullify other query results
remove the exclamation mark `: UpdateAdvertisementPayload`